ReactSkia/modules/platform/libcurl/RSkNetworkingModule.cpp

```diff
@@ -0,0 +1,140 @@
+
+
+}
+
+RSkNetworkingModule::~RSkNetworkingModule() {
+
+};
+
+struct WriteThis {
+  const char *readptr;
+  size_t sizeleft;
+};
+
+
+static size_t WriteCallback(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string*)userp)->append((char*)contents, size * nmemb);
+    return size * nmemb;
+}
+
+
+jsi::Value RSkNetworkingModule::sendRequest(
+  folly::dynamic method,
+  folly::dynamic url,
+  folly::dynamic headers,
+  folly::dynamic data,
+  folly::dynamic responseType,
+  folly::dynamic useIncrementalUpdates,
+  folly::dynamic timeout,
+  folly::dynamic withCredentials) {
+  
+  CURL *curl;
+  CURLcode res;
+  std::string readBuffer;
+  std::string methodString = method.c_str();
+  std::string time = timeout.c_str();
+
+  struct curl_slist *curlListRequestHeader = NULL;
+  struct WriteThis wt;
+
+  if(data["string"].c_str()) {
+      wt.readptr = data["string"].c_str();
+      wt.sizeleft = strlen(data["string"].c_str());
+  } else if(data["formData"].c_str()) {
+      wt.readptr = data["formData"].c_str();
+      wt.sizeleft = strlen(data["formData"].c_str());
+  } else {
+      wt.readptr = data.c_str();
+      wt.sizeleft = strlen(data.c_str());
+  }
+
+  res = curl_global_init(CURL_GLOBAL_DEFAULT);
+  /* Check for errors */
+  if(res != CURLE_OK) {
+     RNS_LOG_ERROR (stderr << "curl_global_init() failed: %s\n" <<
+            curl_easy_strerror(res));
+    return 1;
+  }
+
+  /* get a curl handle */
+  curl = curl_easy_init();
+
+  if(curl != NULL) {
+ 
+      curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+      /*The following code gets executed for a https connection.*/
+
+      if(strstr(url.c_str(),"https") != NULL) {
+            curl_easy_setopt(curl, CURLOPT_SSLENGINE_DEFAULT, 1L);
+            curl_easy_setopt(curl, CURLOPT_CAINFO, CA_CERTIFICATE);
+            curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,1);
+      }
+
+      if(methodString.compare("POST") == 0) { 
+          curl_easy_setopt(curl, CURLOPT_POST, 1L);
+	  for (auto& pair : headers.items()) {
+              // Key is pair.first, value is pair.second
+              string headerKeyValue = pair.first.c_str();
+              const char* parse= ": ";
+              headerKeyValue += parse;
+              headerKeyValue += pair.second.c_str();
+              curlListRequestHeader = curl_slist_append(curlListRequestHeader, headerKeyValue.c_str());
+
+           }
+
+           curl_easy_setopt(curl, CURLOPT_HTTPHEADER, curlListRequestHeader);
+           /* get verbose debug output please */
+           
+	   curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
+	   curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, (long)wt.sizeleft);
+	   curl_easy_setopt(curl, CURLOPT_COPYPOSTFIELDS, wt.readptr);
+       }
+
+      if(methodString.compare("GET") == 0) {
+          curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+          curl_easy_setopt(curl, CURLOPT_WRITEDATA, &readBuffer);
```

Comment:
This should be fine since this readBuffer is sent to WriteCallback when we call curl_easy_perform().
But I don't see  readBuffer being used anywhere.